src/overlays/actors/ovl_En_In/z_en_in.c

```diff
@@ -21,92 +44,1623 @@ const ActorInit En_In_InitVars = {
+                             sAnimations[animIndex].transitionRate);
+    }
+    return ret;
+}
+
+s32 func_808F3178(EnIn* this, GlobalContext* globalCtx) {
+    u8 prevUnk260 = this->unk260;
+    u8 prevUnk261 = this->unk261;
+    u8 tmp;
+
+    tmp = func_8013DB90(globalCtx, &this->unk248, -6.0f);
+    this->unk260 = tmp;
+    if (this->unk260 != 0 && prevUnk260 == 0 && tmp & 0xFF) {
+        Audio_PlayActorSound2(&this->actor, 0x802);
+    }
+    tmp = func_8013DB90(globalCtx, &this->unk254, -6.0f);
+    this->unk261 = tmp;
+    if (this->unk261 != 0 && prevUnk261 == 0 && tmp & 0xFF) {
+        Audio_PlayActorSound2(&this->actor, 0x802);
+    }
+    return 0;
+}
+
+void func_808F322C(EnIn* this, s32 arg1) {
+    if (DECR(this->unk484) == 0) {
+        this->unk482++;
+        if (this->unk482 >= arg1) {
+            this->unk482 = 0;
+            this->unk484 = Rand_S16Offset(0x1E, 0x1E);
+        }
+    }
+}
+
+void func_808F32A0(EnIn* this, GlobalContext* globalCtx) {
+    this->colliderCylinder.dim.pos.x = this->actor.world.pos.x;
+    this->colliderCylinder.dim.pos.y = this->actor.world.pos.y;
+    this->colliderCylinder.dim.pos.z = this->actor.world.pos.z;
+    if (this->unk23D == 0) {
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colCheckCtx, &this->colliderCylinder.base);
+    }
+}
+
+s32 func_808F3310(EnIn* this, GlobalContext* globalCtx) {
+    if ((this->colliderCylinder.base.acFlags & 2)) {
+        this->colliderCylinder.base.acFlags &= ~2;
+    }
+    return 0;
+}
+
+s32 func_808F3334(EnIn* this, GlobalContext* globalCtx) {
+    ActorPlayer* player = PLAYER;
+
+    if (this->colliderJntSph.base.atFlags & 2) {
+        this->colliderJntSph.base.atFlags &= ~2;
+        if (this->colliderJntSph.base.atFlags & 4) {
+            return 0;
+        }
+        Audio_PlayActorSound2(&player->base, 0x83E);
+        func_800B8D98(globalCtx, &this->actor, 3.0f, this->actor.yawTowardsPlayer, 6.0f);
+    }
+    return 1;
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_In_0x808F30B0/func_808F3178.asm")
+s32 func_808F33B8(void) {
+    s32 ret;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_In_0x808F30B0/func_808F322C.asm")
+    if (((ret = gSaveContext.day == 1) && (ret = gSaveContext.time >= 0x3AAA) && (ret = gSaveContext.time < 0x4001)) ||
+        (ret = gSaveContext.day >= 2)) {
+        ret = (gSaveContext.weekEventReg[22] & 1) == 0;
+    }
+    return ret;
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_In_0x808F30B0/func_808F32A0.asm")
+void func_808F3414(EnIn* this, GlobalContext* globalCtx) {
+    ActorPlayer* player = PLAYER;
+    Vec3f sp30;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_In_0x808F30B0/func_808F3310.asm")
+    if (this->unk23D == 0) {
+        this->unk494 = SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    }
+    if (func_8013D5E8(this->actor.shape.rot.y, 0x2710, this->actor.yawTowardsPlayer)) {
```

Comment:
```suggestion
    if (func_8013D5E8(this->actor.shape.rot.y, 10000, this->actor.yawTowardsPlayer)) {
```

Reply:
I can't be sure without decomping `func_8013D5E8`, but I feel that `0x2710` is going to turn out to be an angle (since we are dealing with rot and yaw). In which case I think it would be left in hex.

Reply:
Agreed. I say make it hex for now.